lib/js/components/Outline/OutlineItem/OutlineItem.vue

```diff
@@ -0,0 +1,273 @@
+<template>
+	<div
+		v-ripple
+		class="outlineItem"
+		:class="{
+			'-disabled': isDisabled,
+			'-medium': size === OUTLINE_ITEM_SIZES.M,
+			'-selected': isSelected,
+			'-backgroundNeutral': backgroundColor === OUTLINE_ITEM_BACKGROUND_COLORS.NEUTRAL,
+		}"
+	>
+		<div class="outlineItem__content">
+			<span v-if="index" class="outlineItem__index" :class="{ '-active': isSelected }">
+				{{ index }}.
+			</span>
+			<ds-icon
+				v-if="iconLeft"
+				class="outlineItem__icon"
+				:class="{
+					'-active': isSelected && isSelectedIconsColor,
+				}"
+				:icon="iconLeft"
+				:size="ICON_SIZES.X_SMALL"
+			/>
+			<span>
+				<span class="outlineItem__label" :class="{ '-uppercase': isLabelUppercase }">
+					{{ label }}
+				</span>
+				<span v-if="additionalText" class="outlineItem__additionalText">
+					{{ additionalText }}
+				</span>
+			</span>
+		</div>
+		<div class="outlineItem__rightContent">
+			<div v-if="$slots.default"><slot /></div>
+			<ds-icon
+				v-if="isDone"
+				class="outlineItem__icon -active"
+				:icon="ICONS.FA_CHECK_SOLID"
+				:size="ICON_SIZES.X_SMALL"
+			/>
+			<ds-icon
+				v-else-if="iconRight"
+				class="outlineItem__icon"
+				:class="{
+					'-active': isSelected && isSelectedIconsColor,
+				}"
+				:icon="iconRight"
+				:size="ICON_SIZES.X_SMALL"
+				:rotation="iconRightRotation"
+			></ds-icon>
+		</div>
+	</div>
+</template>
+
+<style scoped lang="scss">
+@import '../../../../styles/settings/radiuses';
+@import '../../../../styles/settings/spacings';
+@import '../../../../styles/settings/colors/tokens';
+@import '../../../../styles/settings/typography/tokens';
+
+.outlineItem {
+	$root: &;
+
+	background-color: $color-neutral-background-weak;
+	border-radius: $radius-s;
+	display: flex;
+	justify-content: space-between;
+	padding: $space-xs;
+
+	&__rightContent,
+	&__content {
+		align-items: center;
+		column-gap: $space-xxs;
+		display: flex;
+	}
+
+	&__index {
+		@include label-l-default-bold;
+
+		color: $color-neutral-text-weak;
+
+		&.-active {
+			color: $color-primary-text;
+		}
+	}
+
+	&__label {
+		@include label-l-default-regular;
+
+		color: $color-neutral-text-heavy;
+
+		&.-uppercase {
+			@include label-l-default-bold-uppercase;
+		}
+	}
+
+	&__additionalText {
+		@include label-m-default-bold;
+
+		color: $color-neutral-text-weak;
+		margin-left: $space-xxxxs;
+	}
+
+	&__icon {
+		color: $color-neutral-icon-weak;
+
+		&.-active {
+			color: $color-primary-icon;
+		}
+	}
+
+	&:hover {
+		background-color: $color-neutral-background-weak-hovered;
+		cursor: pointer;
+	}
+
+	&.-selected {
+		border-radius: $radius-xs $radius-s $radius-s $radius-xs;
+		box-shadow: -2px 0 0 $color-primary-border;
```

Comment:
i don't see shadow in figma, where do you have it?

Reply:
and why we use border token in box-shadow?

Reply:
I think it's bacause we use box-shadow here. I don't know if we can fix this. Maybe @kvas-damian knows.

Reply:
fixed 🚀